### DISCUSSION TITLE
v2.4.0

### DISCUSSION THREAD

**Discussion by crohr** [runs-on/runs-on#115] [Announcements]
Summary: beta windows support, prometheus metrics, disk statistics in workflow logs.

## New features

- Prometheus metrics export, every minutes, at `/metrics` (authenticated with Basic Auth and a new `ServerPassword` CloudFormation parameter).
  - **`runs_on_ec2_instances_total`**, across various labels: `image_id`, `az`, `instance_type`, `instance_lifecycle`, `instance_state`, `repo_full_name`, `runner_id`, `workflow_job_name`, `workflow_job_started`, `workflow_name`.
  - **`runs_on_cloudtrail_events_total`**, across labels `event_name`, for `CreateFleet`, `RunInstances`, and `BidEvictedEvent` events.

  <img width="1417" alt="metrics" src="https://github.com/user-attachments/assets/d44f00c5-46c4-43b7-9164-ddd72eb02ec7">

- Windows support (x64 only for now), with a base image: `image=windows22-base-x64`. Dependencies have to be installed in your workflow steps, or you need to build a custom AMI based on an official Windows 2022 AMI. Current boot time =~ 2min. This will get better.
  <img width="564" alt="image" src="https://github.com/runs-on/runs-on/assets/6114/67da784d-c9dc-40c8-974a-4ef9317c27c8">

- Display disk details in 'Runner Instance' log group.
  - Linux
  <img width="546" alt="image" src="https://github.com/runs-on/runs-on/assets/6114/ae8b1905-3fb8-48a2-8a90-e778ed0cdadc">

  - Windows
  <img width="543" alt="image" src="https://github.com/runs-on/runs-on/assets/6114/5ca17a51-7d42-4516-a23e-17f4ae48d9c3">

## Misc

- Agent rewrite, to handle multi-platform (Windows, see above).
- Windows EC2Launch logs are available at `C:\runs-on\output.log`.
- Ability to specify an alternative public ECR registry for the RunsOn docker image.

## Upgrade

* [Upgrade Guide](https://runs-on.com/guides/upgrade/)
* CloudFormation Versioned template URL: https://runs-on.s3.eu-west-1.amazonaws.com/cloudformation/template-v2.4.0.yaml

<hr /><em>This discussion was created from the release <a href='https://github.com/runs-on/runs-on/releases/tag/v2.4.0'>v2.4.0</a>.</em>

**Comment by MahdiBM** [comment]
Wanted to update to this version but noticed the `pre-release` tag.
Is that intentional?
I would have expected a SemVer that includes prerelease identifiers.

**Reply by crohr** [reply]
Hello, it is intentional. I'm waiting for a few users to test it first, before marking it as final. In RunsOn case a prerelease only means that it is not yet available as the default version when following the instructions from the install guide.
